pkg/k8s/watchers/endpointsynchronizer.go

```diff
@@ -41,6 +41,21 @@ type EndpointSynchronizer struct {
 	Clientset client.Clientset
 }
 
+// NodeAddress is an interface for retrieving the node address this
+// instance of the cilium agent is running on. Abstracted for testing.
+type NodeAddress interface {
+	NodeIP() string
+}
+
+// nodeAddressGetter implements NodeAddress.
+type nodeAddressGetter struct{}
+
+// NodeIP returns the node IP that will be referenced by CiliumEndpoints
+// with endpoints running on this node.
+func (n nodeAddressGetter) NodeIP() string {
+	return node.GetCiliumEndpointNodeIP()
```

Comment:
I think that this can be mocked without need to do this dependency injection

`pkg/node/local_node_store.go`

> newTestLocalNodeStore